src/librustc/ty/maps.rs

```diff
@@ -180,6 +182,20 @@ impl<'tcx> Value<'tcx> for ty::SymbolName {
     }
 }
 
+struct QueryMap<D: QueryDescription> {
+    phantom: PhantomData<D>,
+    map: FxHashMap<D::Key, D::Value>,
```

Comment:
Can we just use `FxHashMap` directly in the macro and perhaps even avoid having a trait for `Key`/`Value` altogether?